analysis_module/mockserver.py

```diff
@@ -0,0 +1,320 @@
+from celery import shared_task
+from sklearn.feature_extraction.text import CountVectorizer
+
+from analysis_module.models import AnalysisModuleRequest
+from core_server.settings import ENDPOINT_NAME
+from .utils import send_callback_url_request
+
+logging.getLogger().setLevel(logging.INFO)
+
+MOCK_GEOLOCATION: List = [
+    {
+        "ent": "Cauca",
+        "offset_start": 0,
+        "offset_end": 0,
+        "geoids": [
+            {
+                "match": "Departamento del Cauca",
+                "geonameid": 3687029,
+                "latitude": 2.5,
+                "longitude": -76.83333,
+                "featurecode": "ADM1",
+                "contrycode": "CO",
+            }
+        ],
+    },
+    {
+        "ent": "Amazonas",
+        "offset_start": 0,
+        "offset_end": 0,
+        "geoids": [
+            {
+                "match": "Amazonas",
+                "geonameid": 3689982,
+                "latitude": -1.16667,
+                "longitude": -71.5,
+                "featurecode": "ADM1",
+                "contrycode": "CO",
+            }
+        ],
+    },
+    {
+        "ent": "Huila",
+        "offset_start": 0,
+        "offset_end": 0,
+        "geoids": [
+            {
+                "match": "Departamento del Huila",
+                "geonameid": 3680692,
+                "latitude": 2.5,
+                "longitude": -75.58333,
+                "featurecode": "ADM1",
+                "contrycode": "CO",
+            }
+        ],
+    },
+    {
+        "ent": "Putumayo",
+        "offset_start": 0,
+        "offset_end": 0,
+        "geoids": [
+            {
+                "match": "Departamento del Putumayo",
+                "geonameid": 3671178,
+                "latitude": 0.5,
+                "longitude": -76.0,
+                "featurecode": "ADM1",
+                "contrycode": "CO",
+            }
+        ],
+    },
+]
+
+
+def get_entries_data(url: str) -> Any:
+    """get data"""
+    response = requests.get(url)
+    entries_data = json.loads(response.text)
+    return entries_data
+
+
+def save_data_local_and_get_url(dir_name: str, client_id: str, data: Any) -> str:
+    """save"""
+    parent_dirpath = f"media/mock_responses/{dir_name}"
+    if not os.path.exists(parent_dirpath):
+        os.makedirs(parent_dirpath)
+
+    filepath = os.path.join(parent_dirpath, f"{client_id}.json")
+    filepath_local = os.path.join("/tmp", filepath)
+
+    with open(filepath_local, "w", encoding="utf-8") as f:
+        f.write(json.dumps(data))
+    return os.path.join(
+        ENDPOINT_NAME, filepath
+    )  # NOTE: this should be handled from external proxy server
+
+
+def get_ngrams(
+    entries: list,
+    ngram_from: int = 1,
+    ngram_to: int = 1,
+    n: int = 10,
+    max_features: int = 20000,
+) -> List[Tuple[str, int]]:
+    vec = CountVectorizer(
+        ngram_range=(ngram_from, ngram_to),
+        max_features=max_features,
+        stop_words="english",
+    ).fit(entries)
```

Comment:
Can't we use `fit_transform` here? @ranjan-stha

Reply:
@k9845 we could have used it but here .fit and .transform are used separately which is also fine.